### PR TITLE
Fix error on update message endpoint

### PR DESCRIPTION
**Pull request by zxenonx**
Fixes https://linear.app/zurichat2/issue/ZUR-77/error-on-update-message-endpoint

### CODE REVIEW

File: backend/endpoints/messages.py [hunk 1]
```diff
@@ -186,12 +186,13 @@ async def update_message(
...
 
-    message.update(payload)
-    message["edited"] = True
+    payload["edited"] = True
     edited_message = await DB.update(
-        MESSAGE_COLLECTION, document_id=message_id, data=message
+        MESSAGE_COLLECTION, document_id=message_id, data=payload
     )
 
+    message.update(payload)
```
Comment (paulinebanye) [inline backend/endpoints/messages.py:194]: I think this should come before payload["edited"] = True

File: backend/endpoints/messages.py [hunk 2]
```diff
@@ -186,12 +186,13 @@ async def update_message(
             detail="You are not authorized to edit this message",
         )
 
-    message.update(payload)
-    message["edited"] = True
+    payload["edited"] = True
```
Comment (paulinebanye) [inline backend/endpoints/messages.py:189]:
```suggestion
    payload["edited"] = True
    message.update(payload)
```

File: backend/endpoints/messages.py (hunk 1 above)
Comment (zxenonx) [reply backend/endpoints/messages.py:194]: @pauline-banye, if we update the message object before setting the edited field to True in the payload, the edited field will still be False in the response.

Comment (paulinebanye) [reply backend/endpoints/messages.py:194]: Ohhhh I forgot about the edited true flag 🤦‍♀️. 
Yeah it's needed beneath the edited true but why all the way underneath the await db.update?

Comment (zxenonx) [reply backend/endpoints/messages.py:194]: Good point.
It's a common practice to declare/initialize variables as close to their usage as possible.
It's not a fixed rule though.
@pauline-banye

Comment (paulinebanye) [reply backend/endpoints/messages.py:194]: Yeah for "readability" tho but it works and all the tests pass. Great job on the fix boss👍🏻

**Review by paulinebanye — APPROVED** [review]
Thanks for the fix @zxenonx

**Pull request merged**